src/DynamoCore/Library/XmlDocumentationExtensions.cs

```diff
@@ -101,34 +103,55 @@ private static string CleanUpDocString(this string text)
         }
 
         private static string GetMemberElement(FunctionDescriptor function,
-            string suffix, XDocument xml)
+            string property, XmlReader xml, string paramName = "")
         {
-            // Construct the entire function descriptor name, including CLR style names
-            string clrMemberName = GetMemberElementName(function);
+            var assemblyName = function.Assembly;
+            var fullyQualifiedName = GetMemberElementName(function);
+            if (!documentNodes.ContainsKey(fullyQualifiedName))
+                LoadDataFromXml(xml);
+
+            // Case for operators.
+            if (assemblyName == null)
+                return String.Empty;
 
-            // match clr member name
-            var match = xml.XPathEvaluate(
-                String.Format("string(/doc/members/member[@name='{0}']/{1})", clrMemberName, suffix));
+            var keyForOverloaded = documentNodes.Keys.
+                        Where(key => key.Contains(function.ClassName + "." + function.FunctionName)).FirstOrDefault();
```

Comment:
Move the string concatenation out of this to avoid getting called too many times:

```csharp
var keyForOverloaded = function.ClassName + "." + function.FunctionName;
if (!documentNodes.ContainsKey(keyForOverloaded))
    keyForOverloaded = string.Empty;
```